### ISSUE TITLE
Promoting version to the same version twice

### ISSUE THREAD

**Issue by aguschin** [iterative/gto#199]
☝️ While investigating, I found two edge cases that are introduced by the way GTO currently works.

#### 1. A version can only be promoted to production once in the same commit. If after this promotion a different version of the same artifact is promoted to the same stage, you can promote the first version again.

For CommitA (sha: 'commit-a) & CommitB (sha: 'commit-b) this script:
```
$ gto register model commit-a --version v0.0.1
$ gto register model commit-b --version v0.0.2

$ gto promote model production commit-a
$ gto promote model production commit-b
$ gto promote model production commit-a
```

Will yield the following git tags:

Commit A: `model@v0.0.1` `model#production#1` `model#production#3`
Commit B: `model@v0.0.2` `model#production#2`

**Proposed solution**: 
We could either show all promotions for that commit, or filter ‘duplicate’ promotions. For Commit A
1. All promotions would show; `v0.0.1 - production` & `v0.0.1 - production` (twice)
2. Only one would be shown `v0.0.1 - production`

#### 2. Promoting a model without registering a new version

For CommitC (sha: 'commit-c'), this script:

```
$ gto promote model production commit-c --skip-registration
```

Will yield only a git tag `model#production#4`, and no version registration tag.

**Proposed solution**
We should still show this promotion if the model column is set to `Promotion` display setting. We should also allow this commit to be a candidate for a version registration if the model column display setting is switched to `Version registrations`.

_Originally posted by @jellebouwman in https://github.com/iterative/studio/issues/3606#issuecomment-1168249521_

**Comment by aguschin** [comment]
Thank you @jellebouwman!

1. It's a bug, I'll fix this.
2. This is to support simple use case when user doesn't want to register SemVers, and care only about promotions. It kinda aligned with "simple intuitive actions should be possible" idea from @dmpetrov: like promotion with `--simple` flag that creates `rf#prod` instead of `rf#prod#1`. Agree with "Proposed solution". In CLI it will look like this:
<img width="867" alt="Screen Shot 2022-06-28 at 18 15 38" src="https://user-images.githubusercontent.com/6797716/176176088-45ecf593-cf16-4520-97e7-e13690a53aca.png">

QQ about "We should also allow this commit to be a candidate for a version registration" - do you mean the History part?
<img width="613" alt="image" src="https://user-images.githubusercontent.com/6797716/176176492-ebd119d7-c044-42eb-8d15-cc4acfdec3c2.png">

If yes, we may skip showing it since in GTO counterpart there is nothing about it. Then it will be aligned. E.g.:
<img width="1037" alt="Screen Shot 2022-06-28 at 18 18 36" src="https://user-images.githubusercontent.com/6797716/176176632-e53b766d-d67f-44d4-aab4-ffd247ac1ab9.png">

The reason for this is that it's not actual registration. That's why I assumed there is no need in showing `registration` action, but show only `promotion` action there.

cc @omesser

**Comment by jellebouwman** [comment]
> QQ about "We should also allow this commit to be a candidate for a version registration" - do you mean the History part?

As I understand, you specify a commit sha, or HEAD, and specifically say that you do not want to register a version at that commit.

If, at a later stage, you _would_ like to register that commit as a version, a registration action could be done, right?

**Comment by aguschin** [comment]
> Alright, thanks for confirming this is a valid case. We will probably not support doing this in Studio at this stage. We will have to be able to show it in the View table however. Good to know!

Yes, it's definitely doesn't need to be supported in Studio now. The default flow (with registration) is cleaner and should be advised.

> If, at a later stage, you would like to register that commit as a version, a registration action could be done, right?

Yes!

**Comment by tapadipti** [comment]
> It's a bug, I'll fix this.

@aguschin So if the user does promote an older version to production again, then what are the expected tags? In Jelle's example above, they are `model#production#1` and `model#production#3`. Will it just be `model#production#1` instead?

**Comment by aguschin** [comment]
@tapadipti 
No, it will be `model#production#3`. But to promote this, you need to add `--force` flag (or `force=True` arg in API). In the first iteration in Studio we can, either:
1. Always use `force=True` and create tag
2. Don't use `force=True` and error out in this case, saying that the model version is already promoted to selected stage.

**Comment by aguschin** [comment]
Seems like this was fixed by #201
Added a regression test for this in #202
